documentsdb api

ArnabChatterjee20k opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.

You can learn more about contributing to appwrite here: https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md

Happy contributing!

-->

## What does this PR do?
-> Example env for documentsdb
```
_APP_CONNECTIONS_DATABASE_DOCUMENTSDB=selfhosted_fra1=mongodb://user:password@mongodb:27017/appwrite,selfhosted_fra2=mongodb://user:password@mongodb:27017/appwrite
_APP_DATABASE_DOCUMENTSDB_KEYS=documentsdb_selfhosted_fra1,documentsdb_selfhosted_fra2
_APP_DATABASE_DOCUMENTSDB_OVERRIDE=documentsdb_selfhosted_fra1
```

-> Patch script working(adding database attribute to the databases table)

[Screencast from 2025-10-07 13-17-08.webm](https://github.com/user-attachments/assets/61c640da-3c6e-49ab-b02d-0a569840aecc)


## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Screenshots may also be helpful.)

## Related PRs and Issues

- (Related PR or issue)

## Checklist

- [ ] Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
- [ ] If the PR includes a change to an API's metadata (desc, label, params, etc.), does it also include updated API specs and example docs?

ArnabChatterjee20k commented on tests/e2e/Services/Databases/DocumentsDB/DatabasesPermissionsMemberTest.php:
it needs changes

ArnabChatterjee20k commented on tests/e2e/Services/Databases/DocumentsDB/DatabasesPermissionsScope.php:
it needs changes

ArnabChatterjee20k commented on tests/e2e/Services/Databases/DocumentsDB/DatabasesPermissionsTeamTest.php:
it needs changes

abnegate committed:
Remove mounts

abnegate committed:
Fix upsert

abnegate committed:
Fix index text checks

abnegate committed:
Fix usage tests

ArnabChatterjee20k committed:
*updated tests
*updated create documents to use the on next callback

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
linting

abnegate committed:
Update DB

ArnabChatterjee20k committed:
linting

ArnabChatterjee20k committed:
updated injections and reverted reduntant changes

ArnabChatterjee20k committed:
added documentsdb in the github actions

ArnabChatterjee20k committed:
updated get endpoint

abnegate committed:
Fix fulltext error

abnegate committed:
Fix storage error

abnegate committed:
Fix sites error

abnegate committed:
Fix domain test

abnegate committed:
Fix storage test

abnegate committed:
Fix spatial tests

abnegate committed:
Merge remote-tracking branch 'origin/1.8.x' into feat-mongodb

# Conflicts:
#	composer.lock

abnegate committed:
Add CI debug

abnegate committed:
Fix workflows init

abnegate committed:
Revert "Add CI debug"

This reverts commit 2d3dea84e32d0d50c844853dda23d92924ec0b63.

abnegate committed:
Fix fts create on array

abnegate committed:
Update db for index validation fix

abnegate committed:
Merge remote-tracking branch 'origin/1.8.x' into feat-mongodb

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated composer

ArnabChatterjee20k committed:
updated database db resolution in the resources itself instead of switch in the worker action

ArnabChatterjee20k committed:
add references docs for the documentsdb

abnegate committed:
Update deps

abnegate committed:
Merge remote-tracking branch 'origin/1.8.x' into feat-mongodb

ArnabChatterjee20k commented on src/Appwrite/Platform/Modules/Databases/Http/DocumentsDB/Logs/XList.php:
```diff
@@ -0,0 +1,142 @@
+            ->callback($this->action(...));
+    }
+
+    public function action(string $databaseId, array $queries, UtopiaResponse $response, Database $dbForProject, Locale $locale, Reader $geodb): void
+    {
```
make sense, extending it via the database route

ArnabChatterjee20k commented on .env:
```diff
@@ -41,6 +41,8 @@ _APP_DB_SCHEMA=appwrite
 _APP_DB_USER=user
 _APP_DB_PASS=password
 _APP_DB_ROOT_PASS=rootsecretpassword
+_APP_DB_ADAPTER_DOCUMENTS=mongodb
+_APP_DOCUMENTS_DB_HOST=mongo
```
keeping this as _APP_DB_HOST_DOCUMENTSDB=mongodb

ArnabChatterjee20k committed:
* removed dbForDocumentsRecords
* using callback based db resolution
* updated env

ArnabChatterjee20k committed:
updated routes and endpoints

ArnabChatterjee20k committed:
revmoved databasse logs xlist action for documentsdb

abnegate committed:
Update database

abnegate committed:
Merge remote-tracking branch 'origin/1.8.x' into feat-mongodb

# Conflicts:
#	composer.json
#	composer.lock

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

abnegate committed:
Pass adapter max ID length to validator

abnegate committed:
Merge branch '1.8.x' into feat-mongodb

ArnabChatterjee20k committed:
updated tests

ArnabChatterjee20k committed:
added patch script for adding database dsn in the databases collection

abnegate committed:
Fix list wrapping UID injections

abnegate committed:
Merge branch 'feat-mongodb' of github.com:appwrite/appwrite into feat-mongodb

ArnabChatterjee20k committed:
* new db pool implementation for connecting to db type
* fixed migration tests

ArnabChatterjee20k committed:
added project region in the get database dsn

abnegate committed:
Fix inject

abnegate commented on src/Appwrite/Migration/Version/V23.php:
```diff
@@ -47,6 +48,16 @@ private function migrateDatabases(): void
         $this->dbForProject->updateDocuments('databases', new Document(['type' => 'legacy']));
-    }
 
+        // database dsn
+        $database = System::getEnv('_APP_DATABASE_NAME', 'appwrite');
+        $namespace = System::getEnv('_APP_DATABASE_SHARED_NAMESPACE', '');
+        $schema = System::getEnv('_APP_DB_HOST', 'mariadb');
+        $dsn = $schema . '://' . $database;
+        if (!empty($namespace)) {
+            $dsn .= '?namespace=' . $namespace;
+        }
```
Let's keep the current format for old DBs, just hostname

abnegate commented on src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Create.php:
```diff
@@ -112,8 +113,10 @@ public function action(string $databaseId, string $collectionId, string $name, ?
             throw new Exception(Exception::DATABASE_NOT_FOUND);
         }
 
+        $dbForDatabase = call_user_func($getDatabaseDB, $database);
```
Let's use newer syntax, `$dbForDatabases = $getDatabasesDB($database);`

ArnabChatterjee20k committed:
* added prefixed channels for realtime channels along with backward compatibility
* added documentsdb tests

abnegate committed:
Fix index validator params

abnegate committed:
Fix roles param

abnegate committed:
Fix tests

abnegate committed:
Merge remote-tracking branch 'origin/1.8.x' into feat-mongodb

# Conflicts:
#	src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Documents/Create.php
#	src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Documents/Delete.php
#	src/Appwrite/Platform/Modules/Databases/Http/TablesDB/Tables/Rows/Delete.php

ArnabChatterjee20k committed:
fix - updated documentsdb sdk namespace to documentsDB

ArnabChatterjee20k committed:
* added waiting time in usage test so that aggregator can work
* added documentsdb usage test

ArnabChatterjee20k committed:
* added getDatabaseDB in stats worker, migration worker
* added prefixed events for multitype db support with backward compat
* added realtime tests, migration tests

ArnabChatterjee20k committed:
updated sdk names and enums for documentsdb

ArnabChatterjee20k committed:
merged changes from 1.8.x

ArnabChatterjee20k committed:
empty commit

ArnabChatterjee20k committed:
linting + merge conflicts resolving

ItzNotABug committed:
update: enums.

ItzNotABug committed:
update: enums.

abnegate committed:
Merge pull request #10628 from appwrite/fix-enums

Update: enums

ArnabChatterjee20k committed:
event generation fix

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into documents-db-api

ArnabChatterjee20k committed:
* fixed stats usage events(hacky fix)
* updated redundant routes(happened during merge conflicts)

ArnabChatterjee20k committed:
addressed comments
* renamed getDatabaseDB to getDatabasesDB
* renamed dbForDatabase to dbForDatabases
* removed call_user_func and using newer callable syntax

ArnabChatterjee20k committed:
* migration fix for tablesdb, legacy with multitype callback
* updated patch script

ArnabChatterjee20k committed:
removed the documentsdb changes from the migration tests

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/feat-mongodb' into documents-db-api

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
reverted docker compose

ArnabChatterjee20k committed:
updated migrations

ArnabChatterjee20k committed:
* updated env to have mongodb support for the tablesdb/legacy
* setAttributeSupport to support context schema and schemaless in mongodb
* updated the connection registers

ArnabChatterjee20k committed:
changed default fallbackForDB to mariadb

ItzNotABug committed:
update: method name.

ItzNotABug committed:
Merge remote-tracking branch 'origin/documents-db-api' into documents-db-api

ArnabChatterjee20k committed:
updated migration

ArnabChatterjee20k committed:
* updated migration tests
* database dsn fetching
* removed dsn from response models

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated migration worker to pass the dsn resolver for the destination database

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated migration

ArnabChatterjee20k committed:
vectordb (in progress)
* pools , env setup
* collections, db endpoints

ArnabChatterjee20k committed:
updated docker compose setup for mongodb which was causing issue for mongo transactions

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated lock file

ArnabChatterjee20k committed:
updated http php for mongodb setup

ArnabChatterjee20k committed:
* updated transactions
* updated tests

ArnabChatterjee20k committed:
reverted docker compose

ArnabChatterjee20k committed:
updated docuemntsdb transaction path names

ArnabChatterjee20k committed:
merged feat-mongodb

ArnabChatterjee20k committed:
updated registers config

ArnabChatterjee20k committed:
reverted longtext resource file

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated deps

ArnabChatterjee20k committed:
flask test fixing via retry

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
updated tests  and index creation

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated migration merge conflicts

ArnabChatterjee20k committed:
updated tests from feat-mongodb branch

ArnabChatterjee20k committed:
Refactor database handling to streamline DSN retrieval for documents and tables databases

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
fixed merge conflicts

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k commented on app/controllers/api/projects.php:
```diff
@@ -51,6 +51,65 @@
 use Utopia\Validator\URL;
 use Utopia\Validator\WhiteList;
 
+function getDatabaseDSN(string $databasetype, $region): string
```
a common method to extract dsn for specific type of database
A very similar to what we used to do for 'database', just this gives the centralised function for each type of multitype db

ArnabChatterjee20k committed:
added the default database type for the transactions context

ArnabChatterjee20k committed:
Merge branch 'documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
added metadata create on database

ArnabChatterjee20k committed:
Merge branch 'documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
updated database dsn for create in the create database

ArnabChatterjee20k committed:
vector db endpoints
* configs and default attributes + indexes
* separate collection creation setup
* fix -> database metadata fix on collection creation

ArnabChatterjee20k committed:
added vector embedding creation + update endpoint

ArnabChatterjee20k committed:
updated method names and sdk namespaces for vectordb

ArnabChatterjee20k committed:
updated tests

ArnabChatterjee20k committed:
updated realtime, create endpoint , resources, registers

ArnabChatterjee20k committed:
Merge branch 'documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
reverted changes

ArnabChatterjee20k committed:
Merge branch 'documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
reverted docker compose

ArnabChatterjee20k committed:
updated composer packages

ArnabChatterjee20k committed:
reverted error tracing

ArnabChatterjee20k committed:
added timestamp mutation test

ArnabChatterjee20k committed:
added transactions for vectordb

ArnabChatterjee20k committed:
linting

ArnabChatterjee20k committed:
added constants for the vector dimension

ArnabChatterjee20k committed:
updated migrations + index finding issue in vectordb

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated composer

ArnabChatterjee20k committed:
updated database type function

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
resolved merge conflicts

ArnabChatterjee20k committed:
updated migration service extractor for csv import/export

ArnabChatterjee20k committed:
skipped operator tests for mongodb

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
updated migrations for vectordb importing/exporting  in csv endpoints

ArnabChatterjee20k committed:
updated migrations

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
updated tests

ArnabChatterjee20k committed:
updated get project dsn to be of the project db type(dedicate, shared) so that databasesdb and projectdb are not in conflict

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
Add VectorDB text embedding creation endpoint and update related tests

ArnabChatterjee20k committed:
updated composer

ArnabChatterjee20k committed:
updated tests condition

ArnabChatterjee20k committed:
updated tests condition

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
fixed enum route

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
Refactor VectorDB Embeddings API and Update Dimension Handling

- Updated the Create.php file to streamline the embedding creation process by removing unnecessary authentication checks and simplifying the parameter structure.
- Changed the parameter from 'documents' to 'texts' for clarity and adjusted the embedding model handling.
- Modified the response model to include error handling for embedding generation.
- Updated the Embedding response model to reflect changes in dimension naming and added error message handling.
- Refactored VectorDBCollection model to replace 'dimensions' with 'dimension' for consistency across the codebase.
- Adjusted all relevant tests to accommodate the new parameter names and response structures, ensuring comprehensive coverage for the updated functionality.

ArnabChatterjee20k committed:
merged 1.8.x

ArnabChatterjee20k committed:
added stats usage for text embeddings

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
updated composer lock

abnegate commented on src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Documents/Upsert.php:
```diff
@@ -77,24 +77,27 @@ public function __construct()
             ->inject('plan')
+            ->inject('transactionState')
+            ->inject('plan')
             ->callback($this->action(...));
```
Duplicate

abnegate commented on src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Update.php:
```diff
@@ -104,7 +105,8 @@ public function action(string $databaseId, string $collectionId, string $name, ?
 
-        $dbForProject->updateCollection('database_' . $database->getSequence() . '_collection_' . $collection->getSequence(), $permissions, $documentSecurity);
+        $dbForDatabases = $getDatabasesDB($database);
+        $dbForDatabases->updateCollection('database_' . $database->getSequence() . '_collection_' . $collection->getSequence(), $permissions, $documentSecurity);
```
Do we need to do this on dbForProject as well, since we have state in both DB for collections?

ArnabChatterjee20k committed:
* added logger in the create text embedding
* aadded error text metric

ArnabChatterjee20k committed:
updated stats

ArnabChatterjee20k committed:
Update app/config/variables.php

Co-authored-by: Jake Barnby <jakeb994@gmail.com>

ArnabChatterjee20k committed:
Update src/Appwrite/Utopia/Database/Validator/Queries/Base.php

Co-authored-by: Jake Barnby <jakeb994@gmail.com>

ArnabChatterjee20k committed:
updated stats

ArnabChatterjee20k committed:
updated database and collection stats

ArnabChatterjee20k committed:
Refactor database metrics handling for DocumentsDB

- Introduced new methods to retrieve metrics for write and read operations based on database type.
- Updated bulk delete, update, upsert, and create actions to use the new metric methods.
- Enhanced document retrieval and update actions to utilize the new metric structure.
- Added support for DocumentsDB metrics in usage statistics and response models.
- Created new response models for DocumentsDB usage and updated existing models to accommodate new metrics.
- Adjusted migration and stats resources to include DocumentsDB metrics.
- Updated tests to validate the new metrics and ensure correct functionality for DocumentsDB usage.

ArnabChatterjee20k committed:
added documentsdb total reads and writes in the project response

ArnabChatterjee20k committed:
updated console client test

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
linting

ArnabChatterjee20k committed:
updated controllers

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
updated stats for vectordb

ArnabChatterjee20k committed:
updated missing skip metrics and tablesdb tests

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
added usage test for the vectordb

ArnabChatterjee20k committed:
fixed missing project usage fields

ArnabChatterjee20k committed:
updated project deletion worker to delete per database

ArnabChatterjee20k committed:
added deletion test for multidb

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/documents-db-api' into vector-db-api

ArnabChatterjee20k committed:
updated delete worker for vectordb and updated tests

abnegate committed:
Merge pull request #10653 from appwrite/vector-db-api

vectordb api endpoints

abnegate commented on src/Appwrite/Platform/Workers/Deletes.php:
```diff
@@ -527,27 +531,45 @@ protected function deleteProject(Database $dbForPlatform, callable $getProjectDB
         $sharedTablesV2 = !$projectTables && !$sharedTablesV1;
+        $databaseDSNKeys = ['database','documentsDatabase','vectorDatabase'];
+
+        $exectionActionPerDatabase = function (string $databaseDSNKey, $callback) use ($getDatabasesDB, $document) {
```
Spelling

ArnabChatterjee20k committed:
Merge remote-tracking branch 'upstream/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
fixed merge conflicts

ArnabChatterjee20k committed:
fixed merge conflicts

ArnabChatterjee20k committed:
linting

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated resources to use the listeners based on the databasetype

ArnabChatterjee20k committed:
added missing injections

ArnabChatterjee20k committed:
synced vectordb

ArnabChatterjee20k committed:
fixed endpoint in migration

ArnabChatterjee20k committed:
linting and fixing resources timeout and validator

ArnabChatterjee20k committed:
commenting mongodb out for now to check the issues on the ci

ArnabChatterjee20k committed:
updated vectordb test

ArnabChatterjee20k committed:
updated migration

ArnabChatterjee20k committed:
updated migration tests for adding endpoint

ArnabChatterjee20k committed:
updated ollama timeout

ArnabChatterjee20k committed:
updated auth param

ArnabChatterjee20k committed:
updated timeout

ArnabChatterjee20k committed:
Merge branch 'new-pool-adapter' into documents-db-api

ArnabChatterjee20k committed:
updetad pools

ArnabChatterjee20k committed:
linting

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
fixed composer lock

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
updated

ArnabChatterjee20k committed:
updated

ArnabChatterjee20k committed:
updated

ArnabChatterjee20k committed:
updated the format specs

ArnabChatterjee20k committed:
add admin permission to create

ArnabChatterjee20k committed:
Merge branch '1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated project and migraiton worker

ArnabChatterjee20k committed:
updated migration

ArnabChatterjee20k committed:
added get databases db in the document logs

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
added admin auth to vector db

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
added admin auth to the index creation

ArnabChatterjee20k committed:
added admin auth to the index creation

ItzNotABug committed:
add: filtering.

ItzNotABug committed:
update: lint.

abnegate committed:
Merge pull request #11310 from appwrite/database-type-filters

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
synced documents db and vectordb with 1.8.x

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/documents-db-api' into documents-db-api

ArnabChatterjee20k committed:
linting

ArnabChatterjee20k committed:
empty

ArnabChatterjee20k committed:
Merge branch '1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated tests

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/documents-db-api' into documents-db-api

ArnabChatterjee20k committed:
updated docker compose

ArnabChatterjee20k committed:
linting and updated timeouts

ArnabChatterjee20k committed:
updated tests

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated composer

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated composer lock

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated tests

ArnabChatterjee20k committed:
updated

ArnabChatterjee20k committed:
Merge remote-tracking branch 'origin/1.8.x' into documents-db-api

ArnabChatterjee20k committed:
updated lock

ArnabChatterjee20k committed:
changed deps so that create document depends on test create collection

ArnabChatterjee20k committed:
updated migration to remove the dependency on dsn at project level attribute

ArnabChatterjee20k committed:
refactor: streamline database DSN handling and remove deprecated attributes

ArnabChatterjee20k committed:
addressed comments

ArnabChatterjee20k committed:
refractored transaction update

ArnabChatterjee20k committed:
update transaction not to start if there is no operation in the logs

abnegate committed:
Merge pull request #11389 from appwrite/multitypedb-with-dedicated-db

ArnabChatterjee20k commented:
closing this pr as the latest pr branch is sync-mongodb with this branch as the base
